src/force/CabanaPD_ForceModels_PMB.hpp

```diff
@@ -251,6 +251,137 @@ auto createForceModel( PMB, Fracture, ParticleType particles,
         delta, K, G0, temp, alpha, temp0 );
 }
 
+template <typename TemperatureType>
+struct ForceModel<PMB, Elastic, DynamicTemperature, TemperatureType>
+    : public ForceModel<PMB, Elastic, TemperatureDependent, TemperatureType>
+{
+    using base_type =
+        ForceModel<PMB, Elastic, TemperatureDependent, TemperatureType>;
+    using base_model = PMB;
+    using fracture_type = Elastic;
+    using thermal_type = DynamicTemperature;
+
+    using base_type::c;
+    using base_type::delta;
+    using base_type::K;
+
+    // Thermal parameters
+    using base_type::alpha;
+    using base_type::temp0;
+    double thermal_coeff;
+    double kappa;
+    double cp;
+
+    // Explicitly use the temperature-dependent stretch.
+    using base_type::thermalStretch;
+
+    // ForceModel(){};
+    ForceModel( const double _delta, const double _K,
+                const TemperatureType _temp, const double _kappa,
+                const double _cp, const double _alpha,
+                const double _temp0 = 0.0 )
+        : base_type( _delta, _K, _temp, _alpha, _temp0 )
+    {
+        set_param( _delta, _K, _kappa, _cp, _alpha, _temp0 );
+    }
+
+    void set_param( const double _delta, const double _K, const double _kappa,
+                    const double _cp, const double _alpha, const double _temp0 )
+    {
+        base_type::set_param( _delta, _K, _alpha, _temp0 );
+
+        kappa = _kappa;
+        cp = _cp;
+        const double d3 = _delta * _delta * _delta;
+        thermal_coeff = 18.0 * _kappa / Kokkos::numbers::pi_v<double> / d3;
+    }
+};
+
+template <typename ParticleType>
+auto createForceModel( PMB, Elastic, ParticleType particles, const double delta,
+                       const double K, const double kappa, const double cp,
+                       const double alpha, const double temp0 )
+{
+    auto temp = particles.sliceTemperature();
+    using temp_type = decltype( temp );
+    return ForceModel<PMB, Elastic, DynamicTemperature, temp_type>(
+        delta, K, temp, kappa, cp, alpha, temp0 );
+}
+
+template <typename TemperatureType>
+struct ForceModel<PMB, Fracture, DynamicTemperature, TemperatureType>
+    : public ForceModel<PMB, Fracture, TemperatureDependent, TemperatureType>
+{
+    using base_type =
+        ForceModel<PMB, Fracture, TemperatureDependent, TemperatureType>;
+    using base_model = typename base_type::base_model;
+    using fracture_type = typename base_type::fracture_type;
+    using thermal_type = DynamicTemperature;
+
+    using base_type::c;
+    using base_type::delta;
+    using base_type::K;
+
+    // Does not use the base bond_break_coeff.
+    using base_type::G0;
+    using base_type::s0;
+    double thermal_coeff;
```

Comment:
The following parameters are not related to bond-breaking:
double thermal_coeff;
double kappa;
double cp;

Perhaps they need a separate header?

Reply:
I don't understand

Reply:
Oh the comment - the header is the file itself. I moved the lines down for clarity